de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/InheritanceTest.java

```diff
@@ -139,6 +140,38 @@ public void singleLevelInheritancePropagationUITest() {
 		waitForAllBuildersAndUiThread();
 		assertEquals("newDocName", bot.tableWithId("tableDocument").cell(0, Document.PROPERTY_DOCUMENTNAME));
 		assertEquals("overriden note", bot.tableWithId("tableDocument").cell(0, Document.PROPERTY_NOTE));
+	}
+	
+	@Test
+	public void inheritanceTypingDragAndDrop() {
+		rename(elementConfiguration, "ec_1");
+		SWTBotTreeItem elementConfiguration2 = addElement(ElementConfiguration.class, conceptPs, configurationTree);
+		rename(elementConfiguration2, "ec_2");
+		
+		SWTBotTreeItem documentEc1 = addElement(Document.class, conceptPs, elementConfiguration);
+		SWTBotTreeItem documentEc2 = addElement(Document.class, conceptPs, elementConfiguration2);
+
+		openEditor(documentEc1);
+		setText(Document.PROPERTY_DOCUMENTNAME, "docFromEc1");
+
+		openEditor(documentEc2);
+		setText(Document.PROPERTY_DOCUMENTNAME, "docFromEc2");
+		
+		// There are no documents yet with the ElementOccurrence
+		assertThat("There is only the documents folder", elementOccurence.getItems(), arrayWithSize(1));
+		
+		// Now drag the first inheritance
+		elementConfiguration.dragAndDrop(elementOccurence);
+		waitForAllBuildersAndUiThread();
+		
+		openEditor(elementOccurence);
+		assertEquals("Received override from EC1", "docFromEc1", bot.tableWithId("tableDocument").cell(0, Document.PROPERTY_DOCUMENTNAME));
+		
+		// now drag the other and verify
+		elementConfiguration2.dragAndDrop(elementOccurence);
+		waitForAllBuildersAndUiThread();
 		
+		openEditor(elementOccurence);
+		assertEquals("Received override from EC2", "docFromEc2", bot.tableWithId("tableDocument").cell(0, Document.PROPERTY_DOCUMENTNAME));

```

Comment:
Let's maybe also check the number of child items after DND?
Also can we emulate add and override actions?

Reply:
1. Well, we could check for the children, but i think that is covered by the unit tests we added. This test i just implemented to have a rough overview that the registration on the extension point and execution of the dropAdapter works as expected.
2. Unfortunately SWTBot only supports The DragAndDrop Move at the moment. Changing it looks to me like a modification in SWTBot code.